app/render/job/shaderjob.h

```diff
@@ -87,6 +88,19 @@ class ShaderJob : public GenerateJob {
     interpolation_.insert(id, interp);
   }
 
+  void SetVertexCoordinates(QVector<float> vertex_coords)
+  {
+    vertex_overrides_.clear();  
+    foreach (float value, vertex_coords) {
+        vertex_overrides_.append(value);
+    }
```

Comment:
This could simply be `vertex_overrides_ = vertex_coords;` and would have the same effect. Also changing the parameter to `const QVector<float> &vertex_coords` is preferable to prevent deep copies.